tests/RocksDBEngine/IndexEstimatorTest.cpp

```diff
@@ -333,12 +344,86 @@ TEST_F(IndexEstimatorTest, test_truncate_logic_2) {
 
   auto expected = currentSeq;
   std::string serialization;
-  est.serialize(serialization, ++currentSeq);
+  est.serialize(serialization, ++currentSeq, format);
   serialization.clear();
   ASSERT_EQ(est.appliedSeq(), expected);
   ASSERT_EQ(1.0, est.computeEstimate());
 
-  est.serialize(serialization, ++currentSeq);
+  est.serialize(serialization, ++currentSeq, format);
   ASSERT_EQ(est.appliedSeq(), expected);
   ASSERT_EQ(1.0, est.computeEstimate());
 }
+
+TEST_F(IndexEstimatorTest, test_serialize_compression) {
+  std::vector<uint64_t> toInsert(10000);
+  constexpr uint64_t seq = 42;
+
+  auto buildEstimator = [&]() {
+    uint64_t i = 0;
+    auto est = std::make_unique<RocksDBCuckooIndexEstimatorType>(2048);
+    std::generate(toInsert.begin(), toInsert.end(), [&i] { return i++; });
+    for (auto it : toInsert) {
+      est->insert(it);
+    }
+
+    est->setAppliedSeq(seq);
+    
+    return est;
+  };
+  
+  auto validateSerializedValue = [&](auto& est, std::string const& serialization) {
+    arangodb::velocypack::StringRef ref(serialization);
+    RocksDBCuckooIndexEstimatorType copy(ref);
+
+    // After serialization => deserialization
+    // both estimates have to be identical
+    EXPECT_EQ(est.nrUsed(), copy.nrUsed());
+    EXPECT_EQ(est.nrCuckood(), copy.nrCuckood());
+    EXPECT_EQ(est.computeEstimate(), copy.computeEstimate());
+    EXPECT_EQ(seq, copy.appliedSeq());
+
+    // Now let us remove the same elements in both
+    bool coin = false;
+    for (auto it : toInsert) {
+      if (coin) {
+        est.remove(it);
+        copy.remove(it);
+      }
+      coin = !coin;
+    }
+
+    // We cannot relibly check inserts because the cuckoo has a random factor
+    // Still all values have to be identical
+
+    EXPECT_EQ(est.nrUsed(), copy.nrUsed());
+    EXPECT_EQ(est.nrCuckood(), copy.nrCuckood());
+    EXPECT_EQ(est.computeEstimate(), copy.computeEstimate());
+  };
+
+  { 
+    // uncompressed
+    auto est = buildEstimator();
```

Comment:
I like this test :+1:

i would have written it a way that I only create the `est` once outside, and do two calls on serialize with different format, then deserialize both, and do a `three-way` equality comparison of each.
To emphasize that they behave exactly the same way, no matter how they are compressed.
This way your estimates slightly differ on the sequence, and in the pseudo random choice you take on inserting equally hashed keys.

However I think the implication of equality on COMPRESSED and UNCOMPRESSED are identical here. So I do not think it is required to add an additional test.